src/primitives/block.h

```diff
@@ -100,6 +103,34 @@ class CBlock : public CBlockHeader
         *((CBlockHeader*)this) = header;
     }
 
+    // return the index of the transaction in this block.  Return -1 if tx is not in this block
+    int find(uint256 hash) const
+    {
+        int nIndex;
+        for (nIndex = 0; nIndex < (int)vtx.size(); nIndex++)
+            if (vtx[nIndex] == *(CTransaction *)this)
+                break;
+        if (nIndex == (int)vtx.size())
+        {
+            nIndex = -1;
+        }
+        return nIndex;
+
+#if 0 // efficient but unnecessary
```

Comment:
why not removing?

Reply:
Find is rarely used if ever, but it remains useful.  The more CPU efficient lookup that is commented out is less memory efficient.  However, if this code ends up being used often, it would make sense to re-enable the alternate implementation.  Why force a future developer to recreate it?

Reply:
I was referring to the code "zeroed" by `#if 0` not the the `find` func.

Reply:
yes, the #if 0 code is a more CPU efficient implementation of find